pandas/tests/indexing/test_categorical.py

```diff
@@ -807,3 +807,31 @@ def test_loc_with_non_string_categories(self, idx_values, ordered):
         result.loc[sl, "A"] = ["qux", "qux2"]
         expected = DataFrame({"A": ["qux", "qux2", "baz"]}, index=cat_idx)
         tm.assert_frame_equal(result, expected)
+
+    @pytest.mark.parametrize(
+        "categories",
+        [
+            pytest.param(["a", "b", "c"], id="str"),
+            pytest.param(
+                [pd.Interval(0, 1), pd.Interval(1, 2), pd.Interval(2, 3)],
+                id="pd.Interval",
+            ),
```

Comment:
Seems these could simply be lists without the pytest.param?

Reply:
Using `pytest.param` with the `id` parameter results in clearer test names:
```
test_categorical.py::TestCategoricalIndex::test_categorical_index_reorder_categories[str] 
test_categorical.py::TestCategoricalIndex::test_categorical_index_reorder_categories[pd.Interval] 
```
Using plain lists results in:
```
test_categorical.py::TestCategoricalIndex::test_categorical_index_reorder_categories[categories0] 
test_categorical.py::TestCategoricalIndex::test_categorical_index_reorder_categories[categories1] 
```